### DISCUSSION TITLE
Is it available to build in MacOS M1?

### DISCUSSION THREAD

**Discussion by wa008** [emscripten-core/emscripten#22482] [General]
When I build c file to html, I meet this problem

```
factorial.c:16:10: fatal error: 'Python.h' file not found
   16 | #include "Python.h"
      |          ^~~~~~~~~~
1 error generated.
emcc: error: '/Users/xxxx/code/emsdk/upstream/bin/clang -target wasm32-unknown-emscripten -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr --sysroot=/Users/xxxx/code/emsdk/upstream/emscripten/cache/sysroot -DEMSCRIPTEN -Werror=implicit-function-declaration -Xclang -iwithsysroot/include/fakesdl -Xclang -iwithsysroot/include/compat factorial.c -c -o /var/folders/_z/n8npvx457n7_9qb_qgxx80jr0000gn/T/emscripten_temp_xmn02nn3/factorial_0.o' failed (returned 1)
```

I found I should install python-dev on mac computer, But I always buid fail. It seems like building issue, I don't know this is my problem or platform problem. I want to know is it essential for me to change a linux or windown platform if I want to use wasm and emscripten smoothly?

install method: python3 -m pip install python-dev-tools --user --upgrad
```
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for libcst
Failed to build libcst
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (libcst)
```

**Comment by sbc100** [comment]
If you want to build code that uses `Python.h` with emscripten then you would first need to build and install the python libraries and headers using emscripten and put them somewhere where emcc can see them.   You can't use the python libraries and headers that come with your host OS.

**Reply by wa008** [reply]
Thanks a lot.
I found where is my python libraries and headers several days ago, which includes "Python.h" file. But I don't know where I show put them in order to make emcc see them, could you provide some advise?

**Reply by sbc100** [reply]
You can either install the headers and libraries into the emscripten sysroot, or you can use `-I` and `-L` flags to point emcc at the headers and libraries you have built.

**Reply by wa008** [reply]
`-I` works for this problem, Thanks very much.

But I met another problem
```
/opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/include/python3.12/pyport.h:586:2: error: "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
```
Most solution after google search are incorrect bit-variation of python, but I found mine is correct(64-bit), it's wired.
```
$ file /opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/bin/python3
/opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/bin/python3: Mach-O 64-bit executable arm64
```

**Reply by sbc100** [reply]
The header file in question looks like it is from homebrew which means it is likely built for your host OS and not for emscripten: /opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/include/python3.12/pyport.h

The headers in `/opt/homebrew` cannot be used with emscripten.

**Reply by wa008** [reply]
Got it. Thanks.
For another solution, what I should do to install the headers and libraries into the emscripten sysroot, I check the document of [emcc](https://emscripten.org/docs/tools_reference/emcc.html#emccdoc), I didn't find related method...

**Reply by sbc100** [reply]
The first step you you likely need to take is to build python (or at least the python libraries and headers) using emscripten.    See  https://stackoverflow.com/questions/44761748/compiling-python-to-webassembly for example.
